utils/generate.ts

```diff
@@ -1,46 +1,43 @@
-import { execSync } from 'child_process'
-import log from 'fancy-log'
 import fs from 'fs'
 import path from 'path'
 
 import { generateAPIClients } from './generator/api-client-generator'
 import { APIModel } from './generator/api-model'
 import {
+  executeCommand,
   generateExportStatement,
-  removeRedundantObjects,
+  generateModelForPreviewVersions,
+  generateModelForStableVersion,
   writeStatementsToFile,
 } from './generator/api-model-generator'
 import { mapEnums2UnionType } from './generator/enum-mapping'
 
 // TODO: Figure out a solution to show commits history in PR description
 async function generateModels(rootPaths: string[]) {
   for (const rootPath of rootPaths) {
-    const models: APIModel[] = fs.readdirSync(rootPath).map((dirname) => {
+    const models: APIModel[] = []
+    const dirnames = fs.readdirSync(rootPath)
+
+    for (const dirname of dirnames) {
       /**
        * API model directory contains both preview and stable version.
-       * Therefore, keep only available API version.
        *
        * Docs: https://github.com/amzn/selling-partner-api-docs/issues/646#issuecomment-825232385
        */
-      const [modelBaseName] = fs.readdirSync(path.resolve(rootPath, dirname))
-      const modelPath = path.resolve(rootPath, dirname, modelBaseName)
-      const outputPath = `src/api-models/${dirname}`
-      const command = `openapi-generator-cli generate -g typescript-axios --additional-properties=supportsES6=true,useSingleRequestParameter=true --type-mappings=set=Array --skip-validate-spec -o ${outputPath} -i ${modelPath}`
-      const model = {
-        modelPath,
-        dirname,
-        outputPath,
-      }
+      const [defaultVersion, ...previewVersions] = fs.readdirSync(path.resolve(rootPath, dirname))
 
-      log.info(`Starting generating ${dirname}`)
-      // TODO: throw an error when command occurs error
-      execSync(command)
-      log.info(`Finished generating ${dirname}`)
+      const stableVersionAPIModel = generateModelForStableVersion(rootPath, dirname, defaultVersion)
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
I think there is a reason it is complaining :)

Maybe try `for await (....)` loop.

Reply:
Resolved: https://github.com/ScaleLeap/selling-partner-api-sdk/pull/147/commits/0b7bc1e6257869ebd82b3d8b4d68835d8b4fbd8e#diff-ba48db8f257b58c9b15436ddd7e5a402ae823cc9b5d5dbfa237903d8ed037b0bR39